core/src/io/anuke/mindustry/ui/dialogs/GameOverDialog.java

```diff
@@ -11,15 +13,48 @@
 
 public class GameOverDialog extends FloatingDialog{
     private Team winner;
+    private Team eliminated;
+    private String mapName;
+    private String author;
+    private int time;
 
     public GameOverDialog(){
         super("$gameover");
         setFillParent(true);
         shown(this::rebuild);
     }
 
+    void resetVars(){
+        if(isShown()){
+            hide();
+        }
+        this.winner = null;
+        this.eliminated = null;
+        this.mapName = null;
+        this.author = null;
+        this.time = -1;
+    }
+
+    public void showEliminated(Team eliminated){
+        resetVars();
+        this.eliminated = eliminated;
+        show();
+    }
+
     public void show(Team winner){
+        resetVars();
+        this.winner = winner;
+        show();
+    }
+
+    public void showRemote(Team winner, String mapName, String author, int time){
+        resetVars();
         this.winner = winner;
+        this.mapName = mapName;
+        if(author != ""){
```

Comment:
Use `equals` for string comparisons instead of `==` or `!=`, e.g. `!"".equals(author)`